src/textual/widgets/_text_area.py

```diff
@@ -1612,6 +1642,17 @@ async def _on_hide(self, event: events.Hide) -> None:
         """Finalize the selection that has been made using the mouse when the widget is hidden."""
         self._end_mouse_selection()
 
+    async def on_click(self, event: events.Click) -> None:
```

Comment:
Maybe this shouldn't use modulus. It doesn't look like vscode wraps the click chain in this way.

Reply:
Experimented with this. For now wrapping works better for us for reasons™.

Without the modulus here, when we get to `chain == 5` and above, the selection goes back to being a cursor and it appears like it has stopped working.

I _think_ VSCode does chaining on the "mouse down" event. This would let us do a better implementation of this and remove a subtle flicker (between chain 2 and 3, the selection disappears briefly as we set the location in the mouse down handler, then the click handler fires later and selects the line). While we only have it on the click event, it's very difficult to replicate their behaviour exactly.